tests/main/xdg-runtime-dir/task.yaml

```diff
@@ -0,0 +1,16 @@
+summary: Ensure that XDG_RUNTIME_DIR is created
+
+details: |
+    Ensure that the XDG_RUNTIME_DIR is created when a snap is launched.
+
+prepare: |
+    echo Testing XDG_RUNTIME_DIR
+    snap install --edge test-snapd-xdg-dir
+    rm -f /run/user/0/snap.test-snapd-xdg-dir
+
+execute: |
+    test ! -e /run/user/0/snap.test-snapd-xdg-dir
+    #   xdgtest returns 255 if the folder doesn't exist,
+    #   making the task fail in that case.
+    snap run test-snapd-xdg-dir || echo "The XDG_RUNTIME_DIR folder doesn't exist inside the environment"
```

Comment:
The `|| echo` bit here seems like it'd eat the a non-zero exit from the snap and let the test continue. Wouldn't it be better to have the test fail immediately in that case?

If we need a diagnostic message for debugging, perhaps the snap's `script.sh` could print it in the error case?

Reply:
Good point too! Done.